presto-main/src/test/java/io/prestosql/sql/planner/TestLogicalPlanner.java

```diff
@@ -937,4 +938,32 @@ public void testRemoveRedundantTopN()
                         node(SortNode.class,
                                 values(ImmutableList.of("t1")))));
     }
+
+    @Test
+    public void testRemoveRedundantDistinctLimit()
+    {
+        String query = "SELECT distinct(c) FROM (SELECT count(*) as c FROM orders) LIMIT 10";
+        assertFalse(
+                searchFrom(plan(query, LogicalPlanner.Stage.OPTIMIZED).getRoot())
+                        .where(isInstanceOfAny(DistinctLimitNode.class))
+                        .matches(),
+                format("Unexpected DistinctLimit node for query: '%s'", query));
+
+        query = "SELECT distinct(c) FROM (SELECT count(*) as c FROM orders GROUP BY orderkey) LIMIT 10";
+        assertPlan(
+                query,
+                output(
+                        node(DistinctLimitNode.class,
+                                anyTree(
+                                        tableScan("orders")))));
+
+        query = "SELECT distinct(id) FROM (VALUES 1, 2, 3, 4, 5, 6) as t1 (id) LIMIT 10";
```

Comment:
can you please extract each test case as separate test method?

Reply:
We have added each test method for the optimizer we implemented. So should we write each pattern of queries  as separate method ?